sklearn/semi_supervised/tests/test_label_propagation.py

```diff
@@ -173,9 +176,8 @@ def test_label_propagation_non_zero_normalizer(LabelPropagationCls):
     X = np.array([[100.0, 100.0], [100.0, 100.0], [0.0, 0.0], [0.0, 0.0]])
     y = np.array([0, 1, -1, -1])
     mdl = LabelPropagationCls(kernel="knn", max_iter=100, n_neighbors=1)
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
         mdl.fit(X, y)
```

Comment:
You need to add to error on warnings otherwise this is not checking that there's no warning. 
you can look at this issue linked in this test to find which warning is expected here (https://github.com/scikit-learn/scikit-learn/issues/9292)